crates/oxc_semantic/src/checker/typescript.rs

```diff
@@ -101,3 +102,18 @@ fn check_simple_assignment_target<'a>(
         }
     }
 }
+
+fn check_array_pattern<'a>(pattern: &ArrayPattern<'a>, ctx: &SemanticBuilder<'a>) {
+    #[derive(Debug, Error, Diagnostic)]
+    #[error("Unexpected type annotation")]
+    #[diagnostic()]
+    struct UnexpectedTypeAnnotation(#[label] Span);
+
+    pattern.elements.iter().for_each(|element| {
```

Comment:
I use plain for loops ever where, shall we keep things consistent?

Reply:
Yes